### PR TITLE
User profile page

### PR DESCRIPTION
**Pull request by KarolDenst**
backend if just hard coded results so frontend works. It will need to be replaced

### CODE REVIEW

File: backend/YouTubeV2.Application/DTO/UserDto.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace YouTubeV2.Application.DTO
+{
+    public record UserDto(string id, string email, string nickname, string name, string surname, double accountBalance, string userType, string avatarImage, int subscriptionsCount);
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/DTO/UserDto.cs]: prob should use one line or new line for every field

File: backend/YouTubeV2.Application/DTO/VideoListDto.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace YouTubeV2.Application.DTO
+{
+    public record VideoListDto(ICollection<VideoMetadataDto> videos);
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/DTO/VideoListDto.cs]: IReadOnlyCollection

File: backend/YouTubeV2.Application/DTO/UserDto.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/DTO/UserDto.cs]: clean up usings everywhere

File: backend/YouTubeV2.Application/DTO/VideoMetadataDto.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+namespace YouTubeV2.Application.DTO
+{
+    public record VideoMetadataDto(string id,
+      string title,
+      string description,
+      string thumbnail,
+      string authorId,
+      string authorNickname,
+      int viewCount,
+      List<string> tags,
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/DTO/VideoMetadataDto.cs]: IReadOnlyCollection insted of List

File: backend/YouTubeV2.Application/DTO/VideoMetadataDto.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace YouTubeV2.Application.DTO
+{
+    public record VideoMetadataDto(string id,
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/DTO/VideoMetadataDto.cs]: id in new line

File: frontend/src/app/app-routing.module.ts [hunk 6]
```diff
@@ -1,9 +1,11 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 import { RegistrationComponent } from '../app/pages/registration/registration.component';
+import { CreatorComponent } from './pages/creator/creator.component';
 
 const routes: Routes = [
   { path: 'register', component: RegistrationComponent },
+  { path: 'creator/:id', component: CreatorComponent },
```
Comment (Marvvvin5) [inline frontend/src/app/app-routing.module.ts:10]: maybe it should be creators/:creatorId

File: frontend/src/app/core/services/user.service.ts [hunk 7]
```diff
@@ -19,4 +19,10 @@ export class UserService {
   registerUser(userForRegistration: UserForRegistrationDTO): Observable<void> {
     return this.httpClient.post<void>(`${this.registrationPageWebAPIUrl}/register`, userForRegistration);
   }
+
+  getUser(id: string): Observable<UserDTO> {
+    let params = new HttpParams().set('id', id);
+    
+    return this.httpClient.get<UserDTO>(`${this.registrationPageWebAPIUrl}/user`, { params: params });
```
Comment (Marvvvin5) [inline frontend/src/app/core/services/user.service.ts:23]: shouldn't we include authentication token?

File: frontend/src/app/core/services/video.service.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  public authenticationStateChanged = this.authenticationStateChangeSubject.asObservable();
+
+  private readonly videoPageWebAPIUrl: string = `${environment.webApiUrl}`;
+
+  constructor(private httpClient: HttpClient) {}
+
+  getUserVideos(id: string): Observable<VideoListDto> {
+    let params = new HttpParams().set('id', id);
+    
+    return this.httpClient.get<VideoListDto>(`${this.videoPageWebAPIUrl}/user/videos`, { params: params });
```
Comment (Marvvvin5) [inline frontend/src/app/core/services/video.service.ts:18]: shouldn't we include authentication token? also is it your responsibility to add logic on be regarding fetching our videos vs fetching videos of someone else?

File: frontend/src/app/core/services/video.service.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+import { Injectable } from "@angular/core";
+import { Observable, Subject } from "rxjs";
+import { environment } from "../../../environments/environment";
+import { VideoListDto } from "../models/video-list-dto";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class VideoService {
+  private authenticationStateChangeSubject = new Subject<boolean>();
```
Comment (Marvvvin5) [inline frontend/src/app/core/services/video.service.ts]: why is it here?

File: frontend/src/app/pages/creator/creator.component.ts [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+import { UserService } from 'src/app/core/services/user.service';
+import { VideoService } from 'src/app/core/services/video.service';
+
+@Component({
+  selector: 'app-creator',
+  templateUrl: './creator.component.html',
+  styleUrls: ['./creator.component.scss']
+})
+export class CreatorComponent {
+  videosPerRow: number = 6;
```
Comment (Marvvvin5) [inline frontend/src/app/pages/creator/creator.component.ts]: why is it like that? cant we use grid (I think it is grid) and choose dynamicaly how many videos per row are visible?

Comment (Marvvvin5) [reply frontend/src/app/pages/creator/creator.component.ts]: the whole logic regarding what video should be in what row will be unncecessary

File: frontend/src/app/pages/creator/creator.component.ts [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+  }
+  
+  getListForRow(row: number): VideoMetadataDto[] {
+    const startIndex = row * this.videosPerRow;
+
+    return this.videos.slice(startIndex, startIndex + this.videosPerRow);
+  }
+
+  public goToUserProfile(id: string): void {
+    console.log(id);
```
Comment (Marvvvin5) [inline frontend/src/app/pages/creator/creator.component.ts]: delete console.log()

File: frontend/src/app/core/services/user.service.ts (hunk 7 above)
Comment (KarolDenst) [reply frontend/src/app/core/services/user.service.ts:23]: we don't have authentication implemented yet

File: frontend/src/app/app-routing.module.ts (hunk 6 above)
Comment (KarolDenst) [reply frontend/src/app/app-routing.module.ts:10]: i don't like it

File: frontend/src/app/core/services/video.service.ts (hunk 8 above)
Comment (KarolDenst) [reply frontend/src/app/core/services/video.service.ts:18]: right now it's noones responsibility

File: frontend/src/app/core/services/video.service.ts (hunk 9 above)
Comment (KarolDenst) [reply frontend/src/app/core/services/video.service.ts]: no idea

Comment (KarolDenst) [reply frontend/src/app/core/services/video.service.ts]: removed it

File: frontend/src/app/pages/creator/creator.component.html [hunk 12]
```diff
@@ -23,22 +23,18 @@ <h2>{{ user.nickname }}</h2>
...
-          <div class="column" *ngFor="let video of getListForRow(row)">              
-            <div class="image-container">
-              <img [src] = "video.thumbnail" class="video">
-              <span class="image-label">{{ video.duration }}</span>
-            </div>
-            <h3 class="noPadding">{{ video.title }}</h3>
-            <p class="noPadding" (click)="goToUserProfile(video.authorId)">{{ video.authorNickname }}</p>
-            <p class="noPadding">{{ video.viewCount }} views</p>
-          </div>
+  <ul *ngIf="videos">
```
Comment (Marvvvin5) [inline frontend/src/app/pages/creator/creator.component.html:26]: what if it's empty?

Comment (KarolDenst) [reply frontend/src/app/pages/creator/creator.component.html:26]: image

File: backend/YouTubeV2/Program.cs [hunk 13]
```diff
@@ -1,5 +1,8 @@
 using Azure.Storage.Blobs;
 using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
```
Comment (Marvvvin5) [inline backend/YouTubeV2/Program.cs]: i think it might not be needed

File: backend/YouTubeV2.Application/Services/SubscriptionsService.cs [hunk 14]
```diff
@@ -28,5 +22,12 @@ public async Task<UserSubscriptionListDTO> GetSubscriptionsAsync(Guid Id, Cancel
                 Select(s => new SubscriptionDTO(new Guid(s.SubscribeeId), _blobImageService.GetProfilePicture(s.Subscribee.Id), s.Subscribee.UserName)).
                 ToListAsync(cancellationToken));
         }
+
+        public async Task<int> GetSubscriptionCount(Guid id, CancellationToken cancellationToken)
+        {
+            var userId = id.ToString();
+
+            return await _context.Subscriptions.Where(s => s.SubscribeeId == userId).CountAsync(cancellationToken);
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Application/Services/SubscriptionsService.cs]: We should check if the userId is a id of a user that is a Creator also add Authorization to the endpoint we dont want anonymous users here. Idk if we can omit where and add lambda to Count but it is something to check

File: backend/YouTubeV2.Api.Tests/SubscriptionsTests.cs [hunk 15]
```diff
@@ -1,15 +1,8 @@
-﻿using FluentAssertions;
```
Comment (Marvvvin5) [inline backend/YouTubeV2.Api.Tests/SubscriptionsTests.cs:1]: Great class! Amazing tests! Either add tests or delete it

File: backend/YouTubeV2.Application/Services/SubscriptionsService.cs (hunk 14 above)
Comment (KarolDenst) [reply backend/YouTubeV2.Application/Services/SubscriptionsService.cs]: I dont like it. This method will be called from user service not from a controller.  I think both authorization and creator check should be checked there not here. As for removing where - good suggestion.

**Review by Marvvvin5 — APPROVED** [review]

**Pull request merged**